tritium-registry/src/main/java/com/palantir/tritium/metrics/registry/AbstractTaggedMetricRegistry.java

```diff
@@ -182,7 +182,7 @@ public final Map<MetricName, Metric> getMetrics() {
                 .forEach((metricName, metric) ->
                         result.put(RealMetricName.create(metricName, tag.getKey(), tag.getValue()), metric)));
 
-        return result.build();
+        return result.buildKeepingLast();
```

Comment:
@carterkozak any concerns with this method only existing in latest [Guava 31.1+ from February 28, 2022](https://github.com/google/guava/releases/tag/v31.1)? I think most of the places using tritium will already have picked up a compatible Guava, just wondering about places like Gradle or other `provided` type class paths that might have older Guava and lead to `NoSuchMethodError`.

https://github.com/google/guava/blob/e4522099b392197a7d2870af634b74fc07d780e7/guava/src/com/google/common/collect/ImmutableMap.java#L619-L622

Reply:
Separately I'm wondering if an error-prone check might be useful in some places to have folks use explicit `buildOrThrow()` or `buildKeepingLast()` as I've seen a few places that made incorrect assumptions.

Reply:
I do t think we should bend over backwards for consumers that disregard the minimum requirements, seven months is long enough that even broken dependency issues should be improbable.

> Separately I'm wondering if an error-prone check might be useful

That’s a great idea, I’ll write it on my flight in a couple hours

Reply:
PR is up :-) https://github.com/palantir/gradle-baseline/pull/2416